message BidRequest {
	required string id= 1;
	repeated Imp 	imp= 2;
	optional Site 	site= 3;
	optional App 	app= 4;
	optional User 	user= 5;
	optional Device device= 6;
	required int32 	test= 7 [default= 0];
	required int32 	tmax= 8;
	required int32  at=9 [default= 2];
	repeated string wseat= 10;
	required Scenario scenario=11; // 流量场景
}

message Scenario{
	required ScenarioType type=1;
	optional string info=2; //场景相关信息, json string  
}

message Imp {
	required string id =1;
	required Banner banner = 2;
	required float 	bidfloor= 3;
}

message Banner {
	required string	id=1;
	optional int32 	w=2;
	optional int32 	h=3;
	optional int32 	wmax=4;
	optional int32 	hmax=5;
	optional int32 	wmin=6;
	optional int32 	hmin=7;
	repeated AdType btype=8;
	repeated AdType wtype=9;
	repeated CreativeAttribute battr=10;
	optional AdPosition pos=11;
	repeated string mimes=12;
	repeated string keywords=13;
	repeated string bwords=14;
}

message Site {
	required string id=1;
	optional string name= 2;
	optional string domain=3;
	repeated ContentCategory cat=4;
	optional ContentCategory sectioncat=5;
	optional ContentCategory pagecat=6;
	required string page= 7;
	optional string ref =8;
	repeated string search= 9;
	required int32 	mobile= 10 [default= 0];
	required Publisher publisher= 11;
	repeated string keywords= 12;
}

message App {
	required string id= 1;
	optional string name= 2;
	required string bundle= 3;
	optional string domain= 4;
	optional string storeurl= 5;
	optional ContentCategory cat=6;
	optional ContentCategory sectioncat=7;
	optional ContentCategory pagecat=8;
	optional string ver= 9;
	required int32 paid= 10;
	required Publisher publisher= 11;
	repeated string keywords= 12;
}

message Publisher {
	required string id=1;
	optional string name=2;
	repeated ContentCategory cat=3;
	required string domain=4;
	optional string type=5;
	optional string slot=6;
}

message Device {
	required string ua=1;
	optional Geo 	geo=2;
	required string ip=3;
	required DeviceType devicetype=4;
	optional string make= 5;
	optional string model= 6;
	optional string hwv= 7;
	optional string os =8;
	optional string osv=9;
	required int32 	w=10;
	required int32 	h=11;
	optional int32 	js=12 [default= 0];
	optional string language=13;
	optional ConnectionType connectiontype=14 [default= ETHERNET];
	optional string imei=15;
	optional string mac=16;
	optional string idfa=17;
	optional string carrier=18;
	optional string androidid=19;
}

message Geo {
	optional float lat=1;
	optional float lon=2;
	required LocationType type=3;
	optional string country=4;
	optional string province=5;
	optional string city=6;
	optional string zip=7;
	optional int32 utcoffset=8 [default= 0];
}

message User {
	required string id=1;
	optional int32 	yob=3;
	optional string gender=4;
	repeated string keywords=5;
	optional Geo 	geo=6;
}

message BidResponse {
	required string id=1;
	repeated SeatBid seatbid=2;
	required string bidid=3;
	optional NoBidReasonCodes nbr=4;
	optional int32 process_time=5;
}

message SeatBid{
	repeated Bid bid= 1;
	required string seat= 2;
}

message Bid {
	required string id=1;
	required string impid=2;
	required float 	price=3;
	required string adm=4;
	required string adomain=5;
	optional string bundle=6;
	required string iurl=7;
	required float 	w=8;
	required float 	h=9;
	optional string adid=10;
	optional string nurl=11;
	optional string cid=12;
	optional string crid=13;
	optional ContentCategory cat=14;
	optional CreativeAttribute attr=15;
	required string curl=16;
	required AdType type=17;
	repeated string extiurl=18;
	optional string action=19; //取值click和install
	required AdmType admtype=20;
}


enum ContentCategory {
	CAT_801 = 801;		// 汽车和交通工具
	CAT_80101 = 80101;	// 小型车
	CAT_80102 = 80102;	// 越野车
	CAT_80103 = 80103;	// 重型车
	CAT_80106 = 80106;	// 自行车及配件
	CAT_80107 = 80107;	// 摩托车及配件
	CAT_80108 = 80108;	// 交通法规
	CAT_80110 = 80110;	// 维修配件
	CAT_802 = 802;		// 投资理财
	CAT_80201 = 80201;	// 保险
	CAT_80203 = 80203;	// 股票
	CAT_80205 = 80205;	// 理财
	CAT_80206 = 80206;	// 基金
	CAT_80207 = 80207;	// 期货
	CAT_80210 = 80210;	// 外汇
	CAT_80212 = 80212;	// 银行
	CAT_80213 = 80213;	// 债券
	CAT_80214 = 80214;	// 信用借贷
	CAT_80215 = 80215;	// 会计审计
	CAT_80216 = 80216;	// 经济新闻
	CAT_803 = 803;		// 母婴
	CAT_80301 = 80301;	// 孕产
	CAT_80302 = 80302;	// 养育
	CAT_80303 = 80303;	// 早教
	CAT_804 = 804;		// 健康医疗
	CAT_80401 = 80401;	// 健康养生
	CAT_80402 = 80402;	// 疾病治疗
	CAT_80403 = 80403;	// 整型整容
	CAT_80404 = 80404;	// 两性
	CAT_80405 = 80405;	// 减肥
	CAT_805 = 805;		// 游戏
	CAT_80501 = 80501;	// 网络游戏
	CAT_80502 = 80502;	// 小游戏
	CAT_80503 = 80503;	// 单机游戏
	CAT_80504 = 80504;	// 网页游戏
	CAT_80505 = 80505;	// 掌上游戏
	CAT_80506 = 80506;	// 手机游戏
	CAT_80507 = 80507;	// 桌上游戏
	CAT_80509 = 80509;	// 游戏周边
	CAT_806 = 806;		// 娱乐影音
	CAT_80601 = 80601; 	// 影视
	CAT_8060101	= 8060101; // 电影
	CAT_8060102	= 8060102; // 电视剧
	CAT_8060103	= 8060103; // 纪录片
	CAT_8060104	= 8060104; // 综艺
	CAT_8060105	= 8060105; // 微电影
	CAT_8060106	= 8060106; // 拍客
	CAT_8060107	= 8060107; // 原创
	CAT_8060108	= 8060108; // 片花
	CAT_8060109	= 8060109; // 广告
	CAT_80602 = 80602;	// 音乐
	CAT_80603 = 80603;	// 动漫
	CAT_80604 = 80604;	// 美图
	CAT_80605 = 80605;	// 星座占卜
	CAT_80606 = 80606;	// 搞笑
	CAT_80607 = 80607;	// 表演艺术
	CAT_80608 = 80608;	// 娱乐八卦
	CAT_80609 = 80609;	// 成人娱乐
	CAT_808 = 808;		// 教育培训
	CAT_80801 = 80801;	// 应试
	CAT_80803 = 80803;	// 培训
	CAT_80805 = 80805;	// 出国
	CAT_80806 = 80806;	// 自考
	CAT_80808 = 80808;	// 社会科学
	CAT_80809 = 80809;	// 自然科学
	CAT_809 = 809;		// 求职招聘
	CAT_80901 = 80901;	// 求职
	CAT_80902 = 80902;	// 招聘
	CAT_810 = 810;		// 家居家电
	CAT_81001 = 81001;	// 家居设计
	CAT_81002 = 81002;	// 装修
	CAT_81003 = 81003;	// 装饰
	CAT_81005 = 81005;	// 家电
	CAT_81007 = 81007;	// 卧室
	CAT_81008 = 81008;	// 厨房
	CAT_81009 = 81009;	// 卫浴
	CAT_811 = 811;		// 房地产
	CAT_81101 = 81101;	// 房产
	CAT_81102 = 81102;	// 出租
	CAT_81103 = 81103;	// 物业
	CAT_81105 = 81105;	// 商业地产
	CAT_812 = 812;		// 体育运动
	CAT_81201 = 81201;	// 足球
	CAT_81202 = 81202;	// 篮球
	CAT_81203 = 81203;	// 排球
	CAT_81204 = 81204;	// 网球
	CAT_81206 = 81206;	// 乒乓球
	CAT_81207 = 81207;	// 田径
	CAT_81208 = 81208;	// 体操
	CAT_81209 = 81209;	// 赛车
	CAT_81210 = 81210;	// 户外运动
	CAT_81211 = 81211;	// 水上运动
	CAT_81212 = 81212;	// 冰雪运动
	CAT_81213 = 81213;	// 搏击运动
	CAT_81214 = 81214;	// 赛事转播
	CAT_81215 = 81215;	// 体育新闻
	CAT_81216 = 81216;	// 其他运动
	CAT_81217 = 81217;	// 健身
	CAT_813 = 813;		// 交通旅游
	CAT_81301 = 81301;	// 酒店住宿
	CAT_81302 = 81302;	// 旅行交通
	CAT_81303 = 81303;	// 旅游服务
	CAT_81304 = 81304;	// 景点
	CAT_81305 = 81305;	// 节假日
	CAT_81306 = 81306;	// 地图
	CAT_81307 = 81307;	// 天气
	CAT_81309 = 81309;	// 特色旅游
	CAT_814 = 814;		// 潮流时尚
	CAT_81401 = 81401;	// 时尚
	CAT_81402 = 81402;	// 美容
	CAT_81403 = 81403;	// 情感
	CAT_81405 = 81405;	// 美发
	CAT_815 = 815;		// 食品饮料
	CAT_81501 = 81501;	// 美食
	CAT_81502 = 81502;	// 零食
	CAT_81503 = 81503;	// 饮料
	CAT_81504 = 81504;	// 烟酒
	CAT_816 = 816;		// IT数码
	CAT_81601 = 81601;	// 手机
	CAT_81602 = 81602;	// 笔记本
	CAT_81603 = 81603;	// 平板
	CAT_81604 = 81604;	// 台式机
	CAT_81605 = 81605;	// 数码相机
	CAT_81606 = 81606;	// 数码影音
	CAT_81607 = 81607;	// 电脑硬件
	CAT_81608 = 81608;	// 企业办公
	CAT_81609 = 81609;	// 软件
	CAT_81610 = 81610;	// 手机app
	CAT_817 = 817;		// 工农业与垂直行业
	CAT_81701 = 81701;	// 农林牧渔
	CAT_81702 = 81702;	// 电子电工
	CAT_81703 = 81703;	// 机械设备
	CAT_81704 = 81704;	// 化工业
	CAT_81705 = 81705;	// 医药生物
	CAT_81706 = 81706;	// 纺织制衣
	CAT_81707 = 81707;	// 金属矿业
	CAT_81709 = 81709;	// 能源
	CAT_81710 = 81710;	// 航天国防
	CAT_81711 = 81711;	// 制造业
	CAT_81712 = 81712;	// 出版业
	CAT_81713 = 81713;	// 建造业
	CAT_81715 = 81715;	// 购物券优惠券
	CAT_81716 = 81716;	// 商品购物（电商）
	CAT_818 = 818;		// 生活服务
	CAT_81804 = 81804;	// 宠物服务
	CAT_81805 = 81805;	// 家政服务
	CAT_81816 = 81816;	// 鲜花绿植
	CAT_81817 = 81817;	// 娱乐休闲
	CAT_81822 = 81822;	// 婚庆、婚纱
	CAT_81823 = 81823;	// 彩票
	CAT_819 = 819;		// 商务服务
	CAT_81901 = 81901;	// 企业服务
	CAT_81902 = 81902;	// 物流仓储
	CAT_81903 = 81903;	// IT服务
	CAT_820 = 820;		// 人文艺术
	CAT_82001 = 82001;	// 摄影
	CAT_82002 = 82002;	// 艺术收藏
	CAT_82003 = 82003;	// 美术设计
	CAT_82005 = 82005;	// 小说电子书
	CAT_82006 = 82006;	// 书籍文学
	CAT_82008 = 82008;	// 人文地理
	CAT_82010 = 82010;	// 工艺制作
	CAT_822 = 822;		// 社交与网络服务
	CAT_82201 = 82201;	// 婚恋交友
	CAT_82202 = 82202;	// SNS、社交
	CAT_82204 = 82204;	// 网络服务
	CAT_82207 = 82207;	// 导航
	CAT_82208 = 82208;	// 网络门户
	CAT_823 = 823;		// 法律、社会、政府
	CAT_82301 = 82301;	// 法律
	CAT_82302 = 82302;	// 时政新闻
	CAT_82303 = 82303;	// 军事
	CAT_82304 = 82304;	// 社会
	CAT_82305 = 82305;	// 政府
	CAT_82306 = 82306;	// 宗教
	CAT_82308 = 82308;	// 民族
	CAT_824 = 824;		// 其他
	CAT_825 = 825;		// 学习参考
	CAT_82501 = 82501;	// IT学习(编程)
	CAT_82502 = 82502;	// 语言学习
	CAT_82503 = 82503;	// 字典、百科
	CAT_82504 = 82504;	// 范文写作
	CAT_82505 = 82505;	// 商务参考（ppt、简历制作等）
	CAT_82506 = 82506;	// 技术参考
	CAT_82507 = 82507;	// 参考工具

}

enum AdType {
	TEXT =1;//文字
	IMAGE=2;//图片
	JS=3;//js
	IFRAME=4;//iframe
}

enum CreativeAttribute {
	ATTR_706 = 706;		// 办公用品
	ATTR_70601 = 70601;	// 办公设备及配件
	ATTR_70602 = 70602;	// 办公耗材
	ATTR_70603 = 70603;	// 文具
	ATTR_70699 = 70699;	// 其他办公用品
	ATTR_711 = 711;		// 车辆
	ATTR_71101 = 71101;	// 微型车
	ATTR_71102 = 71102;	// 小型车
	ATTR_71103 = 71103;	// 紧凑型车
	ATTR_71104 = 71104;	// 中型车
	ATTR_71105 = 71105;	// 中大型车
	ATTR_71106 = 71106;	// 豪华车
	ATTR_71107 = 71107;	// SUV
	ATTR_71108 = 71108;	// MPV
	ATTR_71109 = 71109;	// 跑车
	ATTR_71110 = 71110;	// 商用车
	ATTR_71111 = 71111;	// 特种车辆
	ATTR_71112 = 71112;	// 汽车公司集团
	ATTR_71113 = 71113;	// 汽车实体服务
	ATTR_71114 = 71114;	// 汽车零配
	ATTR_71115 = 71115;	// 汽车电子
	ATTR_71116 = 71116;	// 汽车内饰
	ATTR_71117 = 71117;	// 汽车外饰
	ATTR_71118 = 71118;	// 摩托及配件
	ATTR_71119 = 71119;	// 养护维修改装
	ATTR_71120 = 71120;	// 天然气/石油加油站
	ATTR_71121 = 71121;	// 自行车/助动车/电瓶车
	ATTR_71199 = 71199;	// 其他车辆相关
	ATTR_720 = 720;		// 法律与政府
	ATTR_72001 = 72001;	// 政府、社会组织
	ATTR_72002 = 72002;	// 法律
	ATTR_72003 = 72003;	// 军队
	ATTR_72005 = 72005;	// 政府服务
	ATTR_72006 = 72006;	// 紧急服务
	ATTR_72007 = 72007;	// 公共服务
	ATTR_72008 = 72008;	// 社会福利保障
	ATTR_72099 = 72099;	// 其他法律与政府
	ATTR_722 = 722;		// 房地产
	ATTR_72201 = 72201;	// 出售
	ATTR_72202 = 72202;	// 出租
	ATTR_72203 = 72203;	// 拍卖
	ATTR_72204 = 72204;	// 房地产开发经营
	ATTR_72205 = 72205;	// 房地产中介服务
	ATTR_72206 = 72206;	// 商品房
	ATTR_72207 = 72207;	// 商铺
	ATTR_72208 = 72208;	// 写字楼
	ATTR_72209 = 72209;	// 新房
	ATTR_72210 = 72210;	// 二手房
	ATTR_72211 = 72211;	// 物业管理
	ATTR_72212 = 72212;	// 私人社区
	ATTR_72299 = 72299;	// 其他房地产相关
	ATTR_701 = 701;		// 服装内衣
	ATTR_70101 = 70101;	// 女装
	ATTR_70102 = 70102;	// 女士内衣
	ATTR_70103 = 70103;	// 男装
	ATTR_70104 = 70104;	// 男士内衣
	ATTR_70105 = 70105;	// 亲子装
	ATTR_70106 = 70106;	// 家居服
	ATTR_70107 = 70107;	// 场合礼服
	ATTR_70108 = 70108;	// 袜子
	ATTR_70109 = 70109;	// 泳装
	ATTR_70110 = 70110;	// 运动服
	ATTR_70111 = 70111;	// 服装面料
	ATTR_70199 = 70199;	// 其他服装内衣
	ATTR_719 = 719;		// 互联网与电信
	ATTR_71901 = 71901;	// 安全
	ATTR_71902 = 71902;	// 电子商务平台
	ATTR_71903 = 71903;	// 服务器
	ATTR_71904 = 71904;	// 软件服务
	ATTR_71905 = 71905;	// 通信
	ATTR_71906 = 71906;	// 移动互联
	ATTR_71907 = 71907;	// 硬件服务
	ATTR_71908 = 71908;	// 域名服务
	ATTR_71909 = 71909;	// 云计算
	ATTR_71910 = 71910;	// 支付平台服务
	ATTR_71911 = 71911;	// 有线电视服务
	ATTR_71912 = 71912;	// 卫星服务
	ATTR_71913 = 71913;	// 电话会议
	ATTR_71914 = 71914;	// 网络电话
	ATTR_71999 = 71999;	// 其他互联网与电信
	ATTR_709 = 709;		// 家居生活
	ATTR_70901 = 70901;	// 家装建材
	ATTR_70902 = 70902;	// 家居饰品
	ATTR_70903 = 70903;	// 厨房用具
	ATTR_70904 = 70904;	// 布艺软饰
	ATTR_70905 = 70905;	// 家具
	ATTR_70906 = 70906;	// 床上用品
	ATTR_70907 = 70907;	// 卫浴
	ATTR_70908 = 70908;	// 灯具
	ATTR_70909 = 70909;	// 清洁用品
	ATTR_70910 = 70910;	// 五金工具
	ATTR_70911 = 70911;	// 日用品
	ATTR_70913 = 70913;	// 收纳储物
	ATTR_70915 = 70915;	// 婚庆用品
	ATTR_70917 = 70917;	// 创意礼品
	ATTR_70918 = 70918;	// 宠物/食品及用品
	ATTR_70999 = 70999;	// 其他家居生活
	ATTR_705 = 705;		// 家用电器
	ATTR_70501 = 70501;	// 生活电器
	ATTR_70502 = 70502;	// 大家电
	ATTR_70503 = 70503;	// 影音电器
	ATTR_70504 = 70504;	// 健康护理电器
	ATTR_70505 = 70505;	// 厨卫电器
	ATTR_70506 = 70506;	// 家电配件
	ATTR_70599 = 70599;	// 其他家用电器
	ATTR_721 = 721;		// 金融
	ATTR_72101 = 72101;	// 保险
	ATTR_72102 = 72102;	// 典当
	ATTR_72103 = 72103;	// 股票
	ATTR_72104 = 72104;	// 贵金属
	ATTR_72105 = 72105;	// 理财
	ATTR_72106 = 72106;	// 基金
	ATTR_72107 = 72107;	// 期货
	ATTR_72108 = 72108;	// 期指
	ATTR_72109 = 72109;	// 私募
	ATTR_72110 = 72110;	// 外汇
	ATTR_72111 = 72111;	// 信托
	ATTR_72112 = 72112;	// 银行
	ATTR_72113 = 72113;	// 债券
	ATTR_72114 = 72114;	// 信用借贷
	ATTR_72115 = 72115;	// 会计审计
	ATTR_72116 = 72116;	// 商业金融
	ATTR_72117 = 72117;	// P2P
	ATTR_72199 = 72199;	// 其他金融相关
	ATTR_717 = 717;		// 旅行与旅游
	ATTR_71701 = 71701;	// 出境游
	ATTR_71702 = 71702;	// 国内游
	ATTR_71703 = 71703;	// 周边游
	ATTR_71704 = 71704;	// 特种旅游
	ATTR_71705 = 71705;	// 户外探险
	ATTR_71706 = 71706;	// 酒店服务
	ATTR_71707 = 71707;	// 旅游保险
	ATTR_71708 = 71708;	// 旅行社
	ATTR_71799 = 71799;	// 其他旅行和旅游
	ATTR_707 = 707;		// 美容与个人护理
	ATTR_70701 = 70701;	// 面部护理
	ATTR_70702 = 70702;	// 身体护理
	ATTR_70703 = 70703;	// 彩妆
	ATTR_70704 = 70704;	// 美容工具
	ATTR_70705 = 70705;	// 美发护发
	ATTR_70706 = 70706;	// 男性护理
	ATTR_70707 = 70707;	// 香水香氛
	ATTR_70708 = 70708;	// 美容机构
	ATTR_70709 = 70709;	// 整容
	ATTR_70710 = 70710;	// 减肥
	ATTR_70711 = 70711;	// 丰胸
	ATTR_70712 = 70712;	// 温泉及医疗水疗
	ATTR_70713 = 70713;	// 口腔护理
	ATTR_70714 = 70714;	// 个人清洁
	ATTR_70799 = 70799;	// 其他美容与个人护理
	ATTR_708 = 708;		// 母婴
	ATTR_70801 = 70801;	// 玩具
	ATTR_70802 = 70802;	// 婴幼食品
	ATTR_70803 = 70803;	// 婴幼用品
	ATTR_70804 = 70804;	// 婴童服装配饰
	ATTR_70805 = 70805;	// 孕产妇用品/胎教
	ATTR_70806 = 70806;	// 孕产妇营养品
	ATTR_70899 = 70899;	// 其他母婴相关
	ATTR_723 = 723;		// 农林牧渔
	ATTR_72301 = 72301;	// 畜牧家禽
	ATTR_72302 = 72302;	// 林业
	ATTR_72303 = 72303;	// 园艺
	ATTR_72304 = 72304;	// 作物
	ATTR_72305 = 72305;	// 蔬菜
	ATTR_72306 = 72306;	// 中草药
	ATTR_72307 = 72307;	// 水产养殖与渔业
	ATTR_72308 = 72308;	// 天然纤维
	ATTR_72309 = 72309;	// 特种养殖
	ATTR_72310 = 72310;	// 饲料
	ATTR_72311 = 72311;	// 肥料
	ATTR_72312 = 72312;	// 农用化学品
	ATTR_72313 = 72313;	// 农机工程
	ATTR_72314 = 72314;	// 农业研究
	ATTR_72315 = 72315;	// 农产品检测与分析
	ATTR_72399 = 72399;	// 其他农林牧渔相关
	ATTR_725 = 725;		// 票务
	ATTR_72501 = 72501;	// 交通票务
	ATTR_72502 = 72502;	// 景点门票
	ATTR_72503 = 72503;	// 赛事票务
	ATTR_72504 = 72504;	// 优惠券与折扣
	ATTR_72505 = 72505;	// 娱乐票务
	ATTR_72599 = 72599;	// 其他票务服务
	ATTR_799 = 799;		// 其他
	ATTR_79901 = 79901;	// 其他
	ATTR_718 = 718;		// 商务服务
	ATTR_71801 = 71801;	// 财务会计
	ATTR_71802 = 71802;	// 起名服务
	ATTR_71803 = 71803;	// 法律咨询
	ATTR_71804 = 71804;	// 公司注册
	ATTR_71805 = 71805;	// 广告传媒
	ATTR_71806 = 71806;	// 设计策划
	ATTR_71807 = 71807;	// 人力资源
	ATTR_71808 = 71808;	// 认证服务
	ATTR_71809 = 71809;	// 上市咨询
	ATTR_71810 = 71810;	// 投资担保
	ATTR_71811 = 71811;	// 货运专线
	ATTR_71812 = 71812;	// 物流快递
	ATTR_71813 = 71813;	// 法律咨询
	ATTR_71814 = 71814;	// 招商加盟
	ATTR_71815 = 71815;	// 知识产权
	ATTR_71816 = 71816;	// IT服务
	ATTR_71817 = 71817;	// 喷绘制卡
	ATTR_71818 = 71818;	// 印刷包装
	ATTR_71819 = 71819;	// 礼品定制
	ATTR_71820 = 71820;	// 展会服务
	ATTR_71821 = 71821;	// 翻译速记
	ATTR_71822 = 71822;	// 安装维修
	ATTR_71899 = 71899;	// 其他商务服务
	ATTR_724 = 724;		// 商业与工业
	ATTR_72401 = 72401;	// 包装纸业
	ATTR_72402 = 72402;	// 船舶
	ATTR_72403 = 72403;	// 电子电工
	ATTR_72404 = 72404;	// 纺织
	ATTR_72405 = 72405;	// 环保
	ATTR_72406 = 72406;	// 机床机械
	ATTR_72407 = 72407;	// 机电
	ATTR_72408 = 72408;	// 建筑建材
	ATTR_72409 = 72409;	// 能源产业
	ATTR_72410 = 72410;	// 化学工业
	ATTR_72411 = 72411;	// 五金工具
	ATTR_72412 = 72412;	// 冶金矿产
	ATTR_72413 = 72413;	// 仪器仪表
	ATTR_72414 = 72414;	// 照明
	ATTR_72415 = 72415;	// 航天与国防
	ATTR_72416 = 72416;	// 商业管理
	ATTR_72417 = 72417;	// 商业和工业打印
	ATTR_72418 = 72418;	// 设计与工程
	ATTR_72419 = 72419;	// 食品生产和服务
	ATTR_72420 = 72420;	// 进出口
	ATTR_72421 = 72421;	// 零售贸易
	ATTR_72422 = 72422;	// 科学设备及服务
	ATTR_72423 = 72423;	// 安全设备及服务
	ATTR_72424 = 72424;	// 运输
	ATTR_72499 = 72499;	// 其他商业与工业相关
	ATTR_716 = 716;		// 生活服务
	ATTR_71601 = 71601;	// 搬家服务
	ATTR_71602 = 71602;	// 殡葬服务
	ATTR_71603 = 71603;	// 餐饮服务
	ATTR_71604 = 71604;	// 宠物服务
	ATTR_71605 = 71605;	// 代办服务
	ATTR_71606 = 71606;	// 房产服务
	ATTR_71607 = 71607;	// 礼品服务
	ATTR_71608 = 71608;	// 婚庆服务
	ATTR_71609 = 71609;	// 家政服务
	ATTR_71610 = 71610;	// 洗衣保洁
	ATTR_71611 = 71611;	// 生活配送
	ATTR_71612 = 71612;	// 租车订车
	ATTR_71613 = 71613;	// 缴费充值
	ATTR_71614 = 71614;	// 安装维修
	ATTR_71615 = 71615;	// 物流仓储
	ATTR_71616 = 71616;	// 鲜花绿植
	ATTR_71617 = 71617;	// 安全与保安
	ATTR_71618 = 71618;	// 娱乐休闲
	ATTR_71619 = 71619;	// 征婚交友
	ATTR_71620 = 71620;	// 装修设计
	ATTR_71621 = 71621;	// 宗教/起名/解梦
	ATTR_71699 = 71699;	// 其他生活服务
	ATTR_710 = 710;		// 食品饮料
	ATTR_71001 = 71001;	// 休闲零食
	ATTR_71002 = 71002;	// 奶粉乳品
	ATTR_71003 = 71003;	// 粮油米面/南北干货/调味品
	ATTR_71004 = 71004;	// 生鲜蔬果
	ATTR_71005 = 71005;	// 非酒精饮料
	ATTR_71006 = 71006;	// 酒类饮料
	ATTR_71007 = 71007;	// 茶叶
	ATTR_71008 = 71008;	// 营养保健
	ATTR_71099 = 71099;	// 其他食品饮料
	ATTR_704 = 704;		// 手机数码与消费电子
	ATTR_70401 = 70401;	// 手机
	ATTR_70402 = 70402;	// 相机/摄像机
	ATTR_70403 = 70403;	// 电玩
	ATTR_70404 = 70404;	// 影音产品
	ATTR_70405 = 70405;	// 笔记本
	ATTR_70406 = 70406;	// 平板电脑
	ATTR_70407 = 70407;	// 台式机/一体机/服务器
	ATTR_70408 = 70408;	// 网络设备
	ATTR_70409 = 70409;	// 电脑硬件
	ATTR_70410 = 70410;	// 数码配件
	ATTR_70499 = 70499;	// 其他手机数码与消费电子
	ATTR_703 = 703;		// 体育健身与户外运动
	ATTR_70301 = 70301;	// 体育用品
	ATTR_70302 = 70302;	// 棋牌益智
	ATTR_70303 = 70303;	// 健身器械
	ATTR_70305 = 70305;	// 户外用品
	ATTR_70306 = 70306;	// 体育迷用品
	ATTR_70307 = 70307;	// 体育教学与教练
	ATTR_70308 = 70308;	// 体育场馆设施
	ATTR_70399 = 70399;	// 其他体育健身与户外运动
	ATTR_702 = 702;		// 鞋包配饰
	ATTR_70201 = 70201;	// 男鞋
	ATTR_70202 = 70202;	// 女鞋
	ATTR_70203 = 70203;	// 运动鞋
	ATTR_70204 = 70204;	// 女士配饰
	ATTR_70205 = 70205;	// 男士配饰
	ATTR_70206 = 70206;	// 箱包
	ATTR_70207 = 70207;	// 眼镜及配件
	ATTR_70208 = 70208;	// 皮具及配件
	ATTR_70209 = 70209;	// 帽子围巾手套
	ATTR_70210 = 70210;	// 珠宝/饰品
	ATTR_70212 = 70212;	// 手表
	ATTR_70299 = 70299;	// 其他鞋包配饰
	ATTR_726 = 726;		// 新闻媒体与出版物
	ATTR_72601 = 72601;	// 女性媒体和出版物
	ATTR_72602 = 72602;	// 男性媒体和出版物
	ATTR_72603 = 72603;	// 青少年媒体和出版物
	ATTR_72604 = 72604;	// 婴幼儿媒体和出版物
	ATTR_72605 = 72605;	// 本地新闻媒体和出版物
	ATTR_72606 = 72606;	// 国内新闻媒体和出版物
	ATTR_72607 = 72607;	// 世界新闻与媒体
	ATTR_72608 = 72608;	// 报刊杂志订阅
	ATTR_72609 = 72609;	// 网络媒体
	ATTR_72610 = 72610;	// 新闻媒体
	ATTR_72611 = 72611;	// 出版
	ATTR_72612 = 72612;	// 参考材料
	ATTR_72699 = 72699;	// 其他新闻媒体和出版物
	ATTR_714 = 714;		// 医疗保健
	ATTR_71401 = 71401;	// 药品
	ATTR_71402 = 71402;	// 保健用品
	ATTR_71403 = 71403;	// 医疗服务
	ATTR_71404 = 71404;	// 医疗机构
	ATTR_71405 = 71405;	// 医疗器械及设备
	ATTR_71499 = 71499;	// 其他医疗保健
	ATTR_715 = 715;		// 艺术与娱乐
	ATTR_71501 = 71501;	// 电视
	ATTR_71502 = 71502;	// 电影
	ATTR_71503 = 71503;	// 电子书
	ATTR_71504 = 71504;	// 动漫
	ATTR_71505 = 71505;	// 古董和收藏
	ATTR_71506 = 71506;	// 工艺品
	ATTR_71507 = 71507;	// 乐器及配件
	ATTR_71508 = 71508;	// 珍藏玩具及爱好
	ATTR_71509 = 71509;	// 摄影
	ATTR_71510 = 71510;	// 演出
	ATTR_71511 = 71511;	// 音乐
	ATTR_71512 = 71512;	// 音像
	ATTR_71514 = 71514;	// 彩票
	ATTR_71515 = 71515;	// 成人娱乐
	ATTR_71516 = 71516;	// 书籍
	ATTR_71599 = 71599;	// 其他艺术与娱乐
	ATTR_713 = 713;		// 游戏
	ATTR_71301 = 71301;	// 网络游戏
	ATTR_71302 = 71302;	// 网页游戏
	ATTR_71303 = 71303;	// 单机游戏
	ATTR_71304 = 71304;	// 电子竞技
	ATTR_71305 = 71305;	// 手机游戏
	ATTR_71306 = 71306;	// 小游戏
	ATTR_71307 = 71307;	// 桌上游戏
	ATTR_71308 = 71308;	// 游戏厅
	ATTR_71309 = 71309;	// 游戏周边
	ATTR_71399 = 71399;	// 其他游戏
	ATTR_712 = 712;		// 职业与教育
	ATTR_71201 = 71201;	// IT培训
	ATTR_71202 = 71202;	// 设计培训
	ATTR_71204 = 71204;	// 外语培训
	ATTR_71205 = 71205;	// 文艺培训
	ATTR_71206 = 71206;	// 职业技能培训
	ATTR_71207 = 71207;	// 管理/MBA
	ATTR_71208 = 71208;	// 婴幼早教
	ATTR_71209 = 71209;	// 家教
	ATTR_71210 = 71210;	// 招聘信息
	ATTR_71211 = 71211;	// 求职信息
	ATTR_71212 = 71212;	// 学历教育
	ATTR_71213 = 71213;	// 学校/培训机构
	ATTR_71214 = 71214;	// 出国留学/移民
	ATTR_71215 = 71215;	// 其他求职招聘
	ATTR_71299 = 71299;	// 其他教育培训
	ATTR_727 = 727;		// 成人用品
	ATTR_72701 = 72701;	// 计生用品
	ATTR_72702 = 72702;	// 成人器具
	ATTR_72703 = 72703;	// 情趣内衣
	ATTR_901 = 901;		// 移动应用
	ATTR_90101 = 90101;	// 儿童
	ATTR_90102 = 90102;	// 教育
	ATTR_90103 = 90103;	// 报刊杂志
	ATTR_90104 = 90104;	// 摄影与录像
	ATTR_90105 = 90105;	// 效率
	ATTR_90106 = 90106;	// 生活
	ATTR_90107 = 90107;	// 健康健美
	ATTR_90108 = 90108;	// 旅游
	ATTR_90109 = 90109;	// 音乐
	ATTR_90110 = 90110;	// 体育
	ATTR_90111 = 90111;	// 商务
	ATTR_90112 = 90112;	// 新闻
	ATTR_90113 = 90113;	// 工具
	ATTR_90114 = 90114;	// 娱乐
	ATTR_90115 = 90115;	// 社交
	ATTR_90116 = 90116;	// 美食佳饮
	ATTR_90117 = 90117;	// 财务
	ATTR_90118 = 90118;	// 参考
	ATTR_90119 = 90119;	// 导航
	ATTR_90120 = 90120;	// 医疗
	ATTR_90121 = 90121;	// 图书
	ATTR_90122 = 90122;	// 天气
	ATTR_90123 = 90123;	// 商品指南
	ATTR_90199 = 90199;	// 其他移动应用
}

enum AdPosition {
	UNKNOWN_POSITION=0;			//不确定
	ABOVE_TEH_FOLD=1;	//顶部浮动
	DEPRECATED=2;		//不可见
	BELOW_THE_FOLD=3;	//底部浮动
	HEADER=4;			//头部
	FOOTER=5;			//尾部
	SIDEBAR=6;			//侧边栏
	FULL_SCREEN=7;		//全屏
}
enum ScenarioType {
	CWIFI= 1;
	NETBAR= 2;
	SITES=3;
	APP=4;
}

enum ConnectionType {
	UNKNOWN_TYPE=0;	//不确定
	ETHERNET=1;	//网路运营商
	WIFI=2;		//WIFI
	CELLULAR_NETWORK_UNKNOWN_GENERATION=3;	//手机运营商－不确定
	CELLULAR_NETWORK_2G=4;	//手机运营商－2G
	CELLULAR_NETWORK_3G=5;	//手机运营商－3G
	CELLULAR_NETWORK_4G=6;	//手机运营商－4G
}

enum LocationType {
	GPS =1;				// GPS 经纬度
	IP_ADDRESS=2;			// IP地址
	USER_PROVIDED=3;		// 用户定制	
}

enum DeviceType {
	MOBILE 	= 1;	// Mobile
	PC		= 2;	// PC
}
enum AdmType {
	JSON 	= 1;	// Mobile
	HTML	= 2;	// PC
}

enum NoBidReasonCodes {
	UNKNOWN_ERROR = 0;			// 未知错误
	TECHNICAL_ERROR = 1;		// 竞价出错/拒绝竞价
	INVALID_REQUEST = 2;		// 无效的竞价请求	
}